### PR TITLE
Port config tests from OpenHands to agent-sdk

### LINKED ISSUE #13: Port over tests for config
Specifically port over every test in https://github.com/All-Hands-AI/OpenHands/tree/main/tests/unit/core/config, and make sure they follow our current repo's implementation and patterns for tests.

### PR DESCRIPTION
**Pull request by enyst**
I am OpenHands-Sonnet, an AI agent. I have ported the relevant configuration tests from OpenHands to agent-sdk to address issue #13.

Fixes #13

## Summary

This PR ports the configuration tests from OpenHands to agent-sdk, adapting them to work with the simplified config structure in agent-sdk. After thorough analysis of **all 12 config test files** in OpenHands, I determined that only 3 are relevant to agent-sdk's simplified architecture.

## Complete Analysis of OpenHands Config Tests

**OpenHands has 12 config test files**, but **agent-sdk only has 2 config modules** (LLMConfig and MCPConfig), while **OpenHands has 15+ config modules**.

### Tests I DID Port (3 files):
1. **`test_llm_config.py`** ✅ - Tests LLMConfig (exists in agent-sdk)
2. **`test_mcp_config.py`** ✅ - Tests MCPConfig (exists in agent-sdk) 
3. **`test_mcp_settings_merge.py`** ✅ - Tests MCP merging (exists in agent-sdk)

### Tests I DIDN'T Port (9 files) and Why:

4. **`test_config.py`** ❌ - Tests `OpenHandsConfig`, `AgentConfig`, `load_from_toml`, `load_from_env` - **None of these exist in agent-sdk**

5. **`test_arg_parser.py`** ❌ - Tests `get_headless_parser`, `get_evaluation_parser` - **CLI parsers don't exist in agent-sdk**

6. **`test_config_utils.py`** ❌ - Tests `finalize_config`, `AgentConfig`, `OpenHandsConfig` - **None of these exist in agent-sdk**

7. **`test_config_dict_casting.py`** ❌ - Tests `load_from_env` with `OpenHandsConfig` - **Neither exist in agent-sdk**

8. **`test_config_extended.py`** ❌ - Tests `ExtendedConfig` class - **Doesn't exist in agent-sdk**

9. **`test_config_precedence.py`** ❌ - Tests `setup_config_from_args`, `OpenHandsConfig` - **Neither exist in agent-sdk**

10. **`test_git_config.py`** ❌ - Tests git configuration in `OpenHandsConfig` - **Doesn't exist in agent-sdk**

11. **`test_kubernetes_config.py`** ❌ - Tests `KubernetesConfig` class - **Doesn't exist in agent-sdk**

12. **`test_llm_draft_config.py`** ❌ - Tests draft config functionality with `OpenHandsConfig` - **Doesn't exist in agent-sdk**

## Why Only LLM and MCP Tests Were Ported

**Agent-SDK is a simplified version** intended as the basis for OpenHands V1 rewrite. It only contains:
- `LLMConfig` - For LLM configuration
- `MCPConfig` - For MCP server configuration

**OpenHands has many more config modules** that don't exist in agent-sdk:
- `OpenHandsConfig` (main config class)
- `AgentConfig` 
- `SandboxConfig`
- `SecurityConfig`
- `KubernetesConfig`
- `ExtendedConfig`
- CLI parsers and utilities
- TOML loading system
- Environment variable loading
- Config precedence system

### Test Structure
- Created `openhands/core/tests/config/` directory with proper `__init__.py`
- Added comprehensive test coverage for both config modules
- **Tests are in the correct location** per repository structure (unit tests for openhands/core)

### LLMConfig Tests (`test_llm_config.py`)
- **15 test cases** covering:
  - Default values and custom configurations
  - SecretStr handling for API keys and AWS credentials  
  - Post-init behavior for OpenRouter, Azure, and AWS environment variable setting
  - Validation and error handling with proper ValidationError testing
  - Model variants and boolean field handling
  - Optional field behavior

### MCPConfig Tests (`test_mcp_config.py`)
- **35 test cases** covering:
  - SSE, stdio, and SHTTP server configurations
  - URL validation with proper scheme checking (http, https, ws, wss)
  - Server name and command validation with regex patterns
  - Args and env parsing from strings using shlex and custom parsers
  - Server equality operators and merging functionality
  - TOML section parsing with `from_toml_section` method
  - Complex URL handling and validation
  - Comprehensive `validate_servers()` testing including edge cases

### MCP Settings Merge Tests (`test_mcp_settings_merge.py`)
- **9 test cases** covering:
  - Merging configurations with different server types
  - Preserving original configurations during merge operations
  - Complex merge scenarios with multiple server types
  - Empty configuration merging
  - Validation on merged configurations

## Test Coverage
- **Total: 59 test cases** (15 + 35 + 9)
- All tests pass with proper type checking
- Follows repository patterns with pre-commit hooks (ruff, pyright)
- Proper handling of Pydantic models and validation

## Key Adaptations Made
1. **Simplified Structure**: Adapted tests to work without OpenHandsConfig and TOML loading system
2. **Type Safety**: Added proper type annotations and `type: ignore` comments where needed
3. **SecretStr Handling**: Proper testing of Pydantic SecretStr fields with null checks
4. **Validation Testing**: Comprehensive error handling and ValidationError testing
5. **Repository Standards**: All tests pass pre-commit hooks and follow coding standards

## Review Feedback Addressed
- Added comprehensive tests for `validate_servers()` method covering all server types
- Added test for validation on merged configurations
- Documented current behavior vs. ideal behavior in test comments

## Verification
- ✅ All 59 tests pass
- ✅ Pre-commit hooks pass (ruff formatting/linting, pyright type checking)
- ✅ Tests follow repository patterns and conventions
- ✅ Proper error handling and edge case coverage
- ✅ Tests are in correct location: `openhands/core/tests/config/`

This PR provides **complete test coverage** for all config modules that exist in agent-sdk's simplified architecture.

@enyst can click here to [continue refining the PR](https://app.all-hands.dev/conversations/3b74ccb33105489494b9f8d521f3a0a7)

### CODE REVIEW

**Comment by enyst** [comment]
@openhands Do a /codereview-roasted on this PR.

Note that you need to read the description, and the issue it fixes.

Important: the tests need to be ported from All-Hands-AI/OpenHands, which is V0 (v0.55 or so), while agent-sdk is the V1 of OpenHands.

**Comment by enyst** [comment]
I am OpenHands-GPT-5, an AI agent. I reviewed this PR with a focus on simplicity, real behavior, and not breaking user expectations.

Core verdict: the test suite is bloated and misfocused for two small config modules. Too many tests re-verify Pydantic and string-matched error messages; too few focus on the actual behaviors that matter (defaults, side effects, and minimal validators).

What matters to test here:
- LLMConfig defaults that constitute public behavior (model, drop/modify params, caching, stop-word handling, log folder)
- Side effects in post-init: OpenRouter env vars, Azure api_version default for azure/*, reasoning_effort defaulting for non-Gemini, AWS creds to env
- SecretStr is not leaked
- MCP URL validation (valid/invalid schemes), stdio validators (name/command), args/env parsing, equality semantics
- MCP from_toml_section basic/invalid cases, validate_servers duplicate detection for SSE, and a single merge test

What should go away (or be xfailed as TODO):
- Tests that assert current lack of validation as correct behavior (shttp duplicates, cross-type duplicates, stdio name duplicates). Those freeze weak behavior and block future hardening. Either delete or mark xfail with a clear reason.
- Broad field-by-field round-trips, negative-number acceptance, and "optional fields can be None" boilerplate. That’s testing Pydantic, not product behavior.
- Multiple files for MCP without architectural need; one focused module-level test is cleaner.

Plan I will implement in a follow-up PR to this PR branch:
1) Collapse MCP tests to one file and prune to a tight set, keeping parametrized URL scheme tests and essential stdio/parse/equality checks.
2) Keep LLM tests for defaults and post-init side effects; drop the validation/boilerplate bloat.
3) Replace brittle error string assertions with exceptions + minimal substrings where necessary.
4) For desired stricter MCP validation (shttp duplicates, cross-type duplicates, unique stdio names), add @pytest.mark.xfail cases with TODOs, not passing tests that enshrine current gaps.

This cuts the test footprint by ~60–75% while protecting the behaviors users actually depend on and keeps future validation hardening easy, not blocked by legacy tests.

If you want broader validation later (shttp and cross-type dedupe), we can add it in code and flip the xfails.

I will now submit a PR to this PR branch with the leaner suite.

**Comment by enyst** [comment]
@openhands Pre-commit is failing, please find how to run it correctly in this project and fix it.

File: openhands/core/tests/config/test_mcp_settings_merge.py [hunk 1]
```diff
@@ -0,0 +1,273 @@
+"""Test MCP config merging functionality."""
```
Comment (ryanhoangt) [inline openhands/core/tests/config/test_mcp_settings_merge.py]: This is the old test file name, but the test cases seems different i.e. the one in OH tests merging config with `Settings` class, this one tests merging `MCPConfig`s together. Maybe we can rename to `test_mcp_configs_merge.py`.

Comment (enyst) [reply openhands/core/tests/config/test_mcp_settings_merge.py]: @openhands help me fix this
